pkg/nfd-client/topology-updater/nfd-topology-updater.go

```diff
@@ -209,59 +172,39 @@ func (w *nfdTopologyUpdater) Stop() {
 	}
 }
 
-// connect creates a client connection to the NFD master
-func (w *nfdTopologyUpdater) Connect() error {
-	// Return a dummy connection in case of dry-run
-	if w.args.NoPublish {
-		return nil
-	}
-
-	if err := w.NfdBaseClient.Connect(); err != nil {
+func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha1.ZoneList) error {
+	cli, err := w.apihelper.GetTopologyClient()
+	if err != nil {
 		return err
 	}
-	w.client = pb.NewNodeTopologyClient(w.ClientConn())
-
-	return nil
-}
 
-// disconnect closes the connection to NFD master
-func (w *nfdTopologyUpdater) Disconnect() {
-	w.NfdBaseClient.Disconnect()
-	w.client = nil
-}
-
-// advertiseNodeTopology advertises the topology CR to a Kubernetes node
-// via the NFD server.
-func advertiseNodeTopology(client pb.NodeTopologyClient, zoneInfo v1alpha1.ZoneList, tmPolicy string, nodeName string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-	zones := make([]*v1alpha1.Zone, len(zoneInfo))
-	// TODO: Avoid copying of data to allow returning the zone info
-	// directly in a compatible data type (i.e. []*v1alpha1.Zone).
-	for i, zone := range zoneInfo {
-		zones[i] = &v1alpha1.Zone{
-			Name:      zone.Name,
-			Type:      zone.Type,
-			Parent:    zone.Parent,
-			Resources: zone.Resources,
-			Costs:     zone.Costs,
+	nrt, err := cli.TopologyV1alpha1().NodeResourceTopologies().Get(context.TODO(), w.nodeInfo.nodeName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		nrtNew := v1alpha1.NodeResourceTopology{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: w.nodeInfo.nodeName,
+			},
+			Zones:            zoneInfo,
+			TopologyPolicies: []string{w.nodeInfo.tmPolicy},
 		}
-	}
 
-	topologyReq := &pb.NodeTopologyRequest{
-		Zones:            zones,
-		NfdVersion:       version.Get(),
-		NodeName:         nodeName,
-		TopologyPolicies: []string{tmPolicy},
+		_, err := cli.TopologyV1alpha1().NodeResourceTopologies().Create(context.TODO(), &nrtNew, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create v1alpha1.NodeResourceTopology!:%w", err)
```

Comment:
nit: do we really need to add the API version ?

Reply:
No I don't think so. I was only moving code, trying not to change anything. But I think we can improve this message -> removed (and in another similar place as well).